### ISSUE TITLE
stderr buffer doesn't clear between runs.

### ISSUE THREAD

**Issue by luciansmith** [spyder-ide/spyder#16743]
### Issue Report Checklist

* [x ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ x] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ x] Tried basic troubleshooting (if a bug/error)
    * [x ] Restarted Spyder
    * [x ] Reset preferences with ``spyder --reset``
    * [x ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x ] Tried the other applicable steps from the Troubleshooting Guide
* [ x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

I have a very clear memory of filing or finding this bug in the past, but I cannot now find it again.

When a python program pipes something to stderr, spyder catches this output (and I believe stores it in a file?), and then pipes it to the console after the program finishes or stops.

However, it then doesn't clear that buffer, so from then on, every time it pipes something to the console window, the previous run's messages are still there.

I thought the same thing used to happen with stdout as well, but now Spyder doesn't seem to show stdout at all.

### What steps reproduce the problem?

I don't know of any standard python package function that writes to stderr, but here's a transcript from the console window for a test program I wrote:

In [1]: runfile('C:/Users/Lucian/Desktop/libSEDML-frank/tests/test-data/create_logtest.py', wdir='C:/Users/Lucian/Desktop/libSEDML-frank/tests/test-data', post_mortem=True)

This is a standard error.

In [2]: runfile('C:/Users/Lucian/Desktop/libSEDML-frank/tests/test-data/create_logtest.py', wdir='C:/Users/Lucian/Desktop/libSEDML-frank/tests/test-data', post_mortem=True)

This is a standard error.
This is a standard error.

In [3]: runfile('C:/Users/Lucian/Desktop/libSEDML-frank/tests/test-data/create_logtest.py', wdir='C:/Users/Lucian/Desktop/libSEDML-frank/tests/test-data', post_mortem=True)

This is a standard error.
This is a standard error.
This is a standard error.

In [4]: runfile('C:/Users/Lucian/Desktop/libSEDML-frank/tests/test-data/create_logtest.py', wdir='C:/Users/Lucian/Desktop/libSEDML-frank/tests/test-data', post_mortem=True)

This is a standard error.
This is a standard error.
This is a standard error.
This is a standard error.

### What is the expected output? What do you see instead?

If the buffer were cleared every time, I would get 'This is a standard error.' only once each time.

## Versions

* Spyder version: 5.1.5 None
* Python version: 3.8.6 64-bit
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.6.0               :  7.19.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  21.5.0 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso >=0.7.0;<0.9.0          :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.7.3 (OK)
pygments >=2.0                :  2.7.3 (OK)
pylint >=2.5.0;<2.10.0        :  2.6.0 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.4 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  50.3.2 (OK)
sphinx >=0.6.6                :  3.3.1 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.3 (OK)
textdistance >=4.2.0          :  4.2.0 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  1.0.1 (OK)
zmq >=17                      :  20.0.0 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  3.4.3 (OK)
numpy >=1.7                   :  1.21.3 (OK)
pandas >=1.1.1                :  1.3.4 (OK)
scipy >=0.17.0                :  1.7.1 (OK)
sympy >=0.7.3                 :  1.9 (OK)
```

**Comment by impact27** [comment]
Duplicate of https://github.com/spyder-ide/spyder/issues/13991

**Comment by impact27** [comment]
#14025 will fix the duplication and show stdout for the kernels started by spyder.

**Comment by luciansmith** [comment]
There it is!  I knew I had filed it before, but I could not find the right search term to reveal it.  Thank you!
